### PR TITLE
docs: use variables and templates in the docs

### PR DESCRIPTION
**Pull request by smira**
Only 1.0 docs are updated which will be used as a template for future
documentation versions.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5211)

### CODE REVIEW

File: website/content/v1.0/guides/deploy-metrics-server.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ We can have Kubelet Serving Certificate Approver and metrics-server installed on
 cluster:
   extraManifests:
     - https://raw.githubusercontent.com/alex1989hu/kubelet-serving-cert-approver/main/deploy/standalone-install.yaml
-    - https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+    - https://github.com/kubernetes-sigs/metrics-server/releases/{{< release >}}/download/components.yaml
```
Comment (frezbo) [inline website/content/v1.0/guides/deploy-metrics-server.md:33]: would this work the k8s sigs repo?

File: website/content/v1.0/guides/kubespan.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ cluster:
 
 ### Upgrading an Existing Cluster
 
-In order to enable KubeSpan for an existing cluster, upgrade to the latest v0.14.
+In order to enable KubeSpan for an existing cluster, upgrade to the latest version.
```
Comment (frezbo) [inline website/content/v1.0/guides/kubespan.md:57]: shouldn't we say the latest minimum version maybe? :shrug:

**Review by frezbo — CHANGES_REQUESTED** [review]

File: website/content/v1.0/guides/deploy-metrics-server.md (hunk 1 above)
Comment (smira) [reply website/content/v1.0/guides/deploy-metrics-server.md:33]: oops... too much search/replace, thanks for noticing that

File: website/content/v1.0/guides/kubespan.md (hunk 2 above)
Comment (smira) [reply website/content/v1.0/guides/kubespan.md:57]: this is docs for 1.0, so I would say update to the latest version? I'm not sure if this section should be completely removed?

Comment (frezbo) [reply website/content/v1.0/guides/kubespan.md:57]: yeh, it's fine I guess, don't want someone to use an old version just because the docs said so

**Review by frezbo — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**